core/include/kernel_defines.h

```diff
@@ -187,6 +187,21 @@ extern "C" {
  */
 #define IS_USED(module) IS_ACTIVE(module)
 
+/**
+ * @def         RIOT_VERSION_NUM(major, minor, patch, extra)
+ * @brief       Generates a 64 bit variable of a release version
+ *              Comparisons to this only apply to released branches
```

Comment:
There should also be a big fat `@warning` here, that this is only supposed to be used for version checking in external projects and that it MUST NOT be used in-tree.